### PR TITLE
Ember Upgrade onto fs2.io.tls

### PR DESCRIPTION
**Pull request by ChristopherDavenport**

### CODE REVIEW

File: ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala [hunk 1]
```diff
@@ -7,19 +7,21 @@ import org.http4s.client._
 import cats._
 import cats.implicits._
 import cats.effect._
-import javax.net.ssl.SSLContext
-import scala.concurrent.ExecutionContext
+// import javax.net.ssl.SSLContext
+// import scala.concurrent.ExecutionContext
```
Comment (rossabaker) [inline ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala]: Remove

**Comment by ChristopherDavenport** [comment]
Yes, its currently on the snapshot that is unreleased, however its a major improvement IMO.

**Comment by rossabaker** [comment]
I jumped on this to see if it's ready while I'm preparing releases. All credit for the important work to Chris.

**Comment by rossabaker** [comment]
Still a compile error:

```scala
[error] /home/ross/src/http4s/ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:86:95: type mismatch;
[error]  found   : F[Option[fs2.io.tls.TLSContext]]
[error]  required: F[Option[F[fs2.io.tls.TLSContext]]]
[error]           .fold(Sync[F].delay(TLSContext.system(blocker)).attempt.map(_.toOption))(_.some.pure[F])
[error]                                                                                               ^
```

**Comment by ChristopherDavenport** [comment]
This was changed afterwards due to comments from this PR, now I think the toOption stuff is now unecessary.

**Comment by ChristopherDavenport** [comment]
This is analagous to what we do in blaze. 

https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala#L304

**Comment by rossabaker** [comment]
Oh, right, to delay the construction for those pathological cases where the default throws.  Ugh.  Yeah, good call.

**Pull request merged**